lib/std/Build/Watch.zig

```diff
@@ -237,6 +237,300 @@ const Os = switch (builtin.os.tag) {
             }
         }
     },
+    .windows => struct {
+        const posix = std.posix;
+        const windows = std.os.windows;
+
+        /// Keyed differently but indexes correspond 1:1 with `dir_table`.
+        handle_table: HandleTable,
+        dir_list: std.ArrayListUnmanaged(*Directory),
+        io_cp: ?windows.HANDLE,
+
+        const HandleTable = std.ArrayHashMapUnmanaged(FileId, ReactionSet, FileId.Adapter, false);
+
+        const FileId = struct {
+            volumeSerialNumber: windows.ULONG,
+            indexNumber: windows.LARGE_INTEGER,
+
+            const Adapter = struct {
+                pub fn hash(self: Adapter, a: FileId) u32 {
+                    _ = self;
+                    var hasher = Hash.init(0);
+                    std.hash.autoHash(&hasher, a);
+                    return @truncate(hasher.final());
+                }
+                pub fn eql(self: Adapter, a: FileId, b: FileId, b_index: usize) bool {
+                    _ = self;
+                    _ = b_index;
+                    return a.volumeSerialNumber == b.volumeSerialNumber and a.indexNumber == b.indexNumber;
+                }
+            };
+        };
+
+        const Directory = struct {
+            handle: windows.HANDLE,
+            id: FileId,
+            overlapped: windows.OVERLAPPED,
+            buffer: [64512]u8 align(@alignOf(windows.FILE_NOTIFY_INFORMATION)) = undefined,
+
+            extern "kernel32" fn ReadDirectoryChangesW(
+                hDirectory: windows.HANDLE,
+                lpBuffer: [*]align(@alignOf(windows.FILE_NOTIFY_INFORMATION)) u8,
+                nBufferLength: windows.DWORD,
+                bWatchSubtree: windows.BOOL,
+                dwNotifyFilter: windows.DWORD,
+                lpBytesReturned: ?*windows.DWORD,
+                lpOverlapped: ?*windows.OVERLAPPED,
+                lpCompletionRoutine: windows.LPOVERLAPPED_COMPLETION_ROUTINE,
+            ) callconv(windows.WINAPI) windows.BOOL;
```

Comment:
I should move this back to `std.os.windows.kernel32`?

Reply:
Yeah. I only removed it because nothing used it at the time.